### ISSUE TITLE
Unable to reconstruct any scenes after fine-tuning

### ISSUE THREAD

**Issue by liu123456-shi** [facebookresearch/vggt#290]
Hello, thank you for your excellent work!
I used a small portion of images from a scene in co3d to fine-tune the VGGT test. After several rounds of training, the model was unable to reconstruct any scenes, as shown in the figure below.

<img width="950" height="595" alt="Image" src="https://github.com/user-attachments/assets/38cf9e9f-3ba4-4f9b-98e1-5d450e2bbbe3" />

**Comment by jytime** [comment]
What's the id of this sequence? And have you tried this in our hugging face demo?

**Comment by liu123456-shi** [comment]
Hello, thank you for your reply.
I used your publicly available model.pt to fine-tune it in a scenario with only a few images.
The loss trend during fine-tuning training appears to be normal.
However, during inference, I found that the model's inference results were completely wrong, and it did not even reconstruct any information.
I just noticed that when loading the model, I used  
model.load_state_dict(checkpoint[‘model’], strict=False),  
which means I did not include the point cloud head and tracking head parameters during fine-tuning. So, I would like to ask: are these parameters randomly generated during inference, leading to the failure of the final reconstruction?

**Comment by jytime** [comment]
It is fine, the point cloud head and tracking head will not affect the 3D structure, as long as you are using the results from depth head + camera head, i.e., our default demo setting.

**Comment by liu123456-shi** [comment]
However, I only fine-tuned the parameters of the depth map head with a few images, which resulted in the final model being unusable.
First, there were no point cloud head and tracking head parameters in the model.
Ignoring these parameters resulted in a very poor reconstruction of the model.

**Comment by jytime** [comment]
That's not the case. Training without point cloud head and tracking head will lead to a slightly worse result but will not destroy your training. The result attached seems to indicate a totally broken model. I would suspect there is some bug somewhere in your case.

**Comment by zhangshuoneu** [comment]
Are the train and test images differently? Maybe model has overfitted to train images so that losing generalization ability.

**Comment by liu123456-shi** [comment]
Yes, I also think that the model may have degraded. However, I used the source code directly to fine-tune the dataset.
Now I load the model using the following method:

model.load_state_dict(torch.load(“model/model.pt”))
model.load_state_dict(torch.load(‘training/logs/exp002/ckpts/checkpoint.pt’), strict=False)

The final result is normal. Should the model be loaded and inferred in this way after fine-tuning?

**Comment by ShikangWen2000** [comment]
Yep, I also found this problem. After fine-tuning the pretrained model in the co3d dataset (hydrant class), the performance is totally destroyed.

**Comment by liu123456-shi** [comment]
I don't know why, but even when using the training set during fine-tuning, the effect still drops. There will be a serious stratification phenomenon. May I ask if there is any problem with the data processing

**Comment by jytime** [comment]
I guess it is not the case as someone shared they have been able to almost reproduce VOGT’s performance using the released training code and training from scratch. How many batches are u using now?

**Comment by liu123456-shi** [comment]
It seems that by freezing the main network architecture and fine-tuning the depth head, the performance of the model will be reduced.
Because I think that if the main network is frozen and the training set is used as the test set when fine-tuning the depth head, the model's performance will still decline significantly.
The batch sizes I use are all parameters given in the source code. Local tests were also conducted, and there were no issues with the scale and format of the data.

**Comment by jytime** [comment]
just ran a job with aggregator frozen and a lr of 1e-5, which seems to work for me

**Comment by liu123456-shi** [comment]
I just tried the parameter Settings of 1e-5.
The dataset used was an apple scene in co3D for fine-tuning. However, when training reached 4 epochs, Loss/train_loss_conf_depth became negative, and loss also began to oscillate.

INFO 2025-07-30 15:12:56,262 general.py: 117: Train Epoch: [4][     37/1000000] | Batch Time: 1.1434 (1.1136) | Data Time: 0.0047 (0.0494) | Mem (GB): 30.0000 (28.7368) | Time Elapsed: 00d 00h 09m | Loss/train_loss_objective: -0.1992 (-0.1430) | Loss/train_loss_camera: 0.0067 (0.0120) | Loss/train_loss_T: 0.0041 (0.0063) | Loss/train_loss_R: 0.0013 (0.0020) | Loss/train_loss_FL: 0.0026 (0.0074) | Loss/train_loss_conf_depth: -0.2564 (-0.2287) | Loss/train_loss_reg_depth: 0.0212 (0.0229) | Loss/train_loss_grad_depth: 0.0023 (0.0029) | Grad/aggregator: 0.0000 (0.0000) | Grad/depth: 14.6176 (24.0991) | Grad/camera: 1.2850 (1.2092)
INFO 2025-07-30 15:12:56,886 general.py: 117: Train Epoch: [4][     38/1000000] | Batch Time: 0.6239 (1.1010) | Data Time: 0.0034 (0.0482) | Mem (GB): 30.0000 (28.7692) | Time Elapsed: 00d 00h 09m | Loss/train_loss_objective: 0.0393 (-0.1397) | Loss/train_loss_camera: 0.0286 (0.0123) | Loss/train_loss_T: 0.0184 (0.0065) | Loss/train_loss_R: 0.0060 (0.0021) | Loss/train_loss_FL: 0.0085 (0.0074) | Loss/train_loss_conf_depth: -0.1408 (-0.2271) | Loss/train_loss_reg_depth: 0.0338 (0.0231) | Loss/train_loss_grad_depth: 0.0030 (0.0029) | Grad/aggregator: 0.0000 (0.0000) | Grad/depth: 22.3516 (24.0543) | Grad/camera: 1.2379 (1.2099)
INFO 2025-07-30 15:12:57,698 general.py: 117: Train Epoch: [4][     39/1000000] | Batch Time: 0.8128 (1.0938) | Data Time: 0.0033 (0.0471) | Mem (GB): 30.0000 (28.8000) | Time Elapsed: 00d 00h 09m | Loss/train_loss_objective: -0.2128 (-0.1410) | Loss/train_loss_camera: 0.0106 (0.0123) | Loss/train_loss_T: 0.0075 (0.0065) | Loss/train_loss_R: 0.0013 (0.0021) | Loss/train_loss_FL: 0.0036 (0.0073) | Loss/train_loss_conf_depth: -0.2866 (-0.2282) | Loss/train_loss_reg_depth: 0.0176 (0.0230) | Loss/train_loss_grad_depth: 0.0032 (0.0029) | Grad/aggregator: 0.0000 (0.0000) | Grad/depth: 23.4490 (24.0392) | Grad/camera: 1.5406 (1.2182)

**Comment by jytime** [comment]
ah I finally get this... Please note that train_loss_conf_depth should definitely be negative after convergence. Generally the model will have reasonable performance if the train_loss_conf_depth becomes -0.6 to -0.8.

https://github.com/facebookresearch/vggt/issues/246

You can check the loss detail in the paper about why it will become negative.

**Comment by jytime** [comment]
One more point, the default config was for debugging, and setting the "debug" flag in dataset config as True. This means only the "apple" category will be used for training under that config. That could explain why the performance drop was reported. I have updated the default config now... 

https://github.com/facebookresearch/vggt/commit/e56963328b7476e615ce8dda9164d381f8dc07a3
